src/ir_Samsung.cpp

```diff
@@ -426,12 +426,16 @@ void IRSamsungAc::setRaw(const uint8_t new_code[], const uint16_t length) {
 }
 
 void IRSamsungAc::on(void) {
+  if (getPower())
```

Comment:
This approach works if and only if the initial internal state (`remote_state`) is in the same state of the actual air conditioner. 
For example, assume `stateReset()` sets the internal state to Power On, but the actual air conditioner unit is Power Off.
Doing a `on()` or `setPower(true)` won't cause the class/library to send the special "on" message. Vice-versa for Power Off etc.

Hence it's safest to always send the special message to ensure we are in sync with the a/c unit.

We could change `_sendpower = false;` to `_sendpower = true;` at [line 312](https://github.com/crankyoldgit/IRremoteESP8266/blob/master/src/ir_Samsung.cpp#L312), then this would work.

Reply:
Maybe we could add a force flag to power methods in order to skip the check and ensure the sync with the ac. Also yeah, that should definitely be changed

Reply:
As I don't have access to one of these a/c units, what is the down-side of always sending the special power control message each time `on()` or `off()` is called? e.g. the current implementation

Reply:
The unit beeps two times in a row so it seems like some sort of error code, not a big problem

Reply:
That's what I figured might be the case, but it was best to be sure.
If we want to reduce the second beep, assuming you are controlling it via the `IRac` class, we could incorporate the previous state and do the "if it is already on/off, don't send the resepective on/off message" there.

IIRC, the current implementation creates a new samsung ac class object each time and thus any change we make here won't save that message being sent. i.e. If you are using the `IRac` class (like IRMQTTServer does) then we should make the change in the `samsung()` method in IRac.cpp.

Reply:
No, I'm using the IRSamsungAc class so I tought that was the correct place to make that change